test/libsolutil/JSON.cpp

```diff
@@ -33,6 +33,48 @@ namespace solidity::util::test
 
 BOOST_AUTO_TEST_SUITE(JsonTest, *boost::unit_test::label("nooptions"))
 
+BOOST_AUTO_TEST_CASE(json_types)
+{
+	auto check = [](Json::Value value, string const& expectation) {
+		BOOST_CHECK(jsonCompactPrint(value) == expectation);
+	};
+
+	Json::Value value;
```

Comment:
@chriseth added tests here to show what type corresponds to what output, `{} == Json::Value() == Json::nullValue`.